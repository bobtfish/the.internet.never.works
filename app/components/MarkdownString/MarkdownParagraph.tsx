import { splitParagraphs } from './Parser'
import { MarkdownString } from './MarkdownString'

export function MarkdownParagraph({ markdown }: { markdown: string | React.ReactNode[] }) {
    const paras = splitParagraphs(markdown)
    return <div>{paras.map((para) => {
        // eslint-disable-next-line react/jsx-key
        return <p><MarkdownString markdown={para} /></p>
    }, markdown)}</div>
}