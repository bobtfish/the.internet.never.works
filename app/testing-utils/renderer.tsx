import testRenderer, { ReactTestRenderer, ReactTestRendererJSON } from 'react-test-renderer';
import { MantineProvider } from '@mantine/core';

function TestInnerContents({ children }: { children: React.ReactNode }) {
  return <div id="testInnerContents">{children}</div>
}

function create(ui: React.ReactNode) {
  const tr = testRenderer.create(<MantineProvider><TestInnerContents>{ui}</TestInnerContents></MantineProvider>)
  const tree = tr.toJSON() || [];
  const innerTree = ((tree as ReactTestRendererJSON[]).find((node) => node.type === 'div' && node.props.id === 'testInnerContents')||{children:[]}).children || []
  Object.defineProperty(tr, 'toJSON', {
    writable: true,
    value: () => { return innerTree }
  })  
  return tr
}

const act = testRenderer.act

const renderer = { create, act }

export default renderer

export function getInnerElement(r: ReactTestRenderer) {
  return r.root.findByType(TestInnerContents).props.children;
}