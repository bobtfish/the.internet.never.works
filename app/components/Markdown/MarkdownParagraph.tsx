import { Text } from '@mantine/core'
import cx from 'clsx'
import { splitParagraphs } from './Parser'
import { MarkdownString } from './MarkdownString'
import type { MarkdownParagraphProps } from './types'
import classes from './MarkdownParagraph.module.css'

export function MarkdownParagraph ({
  className,
  markdown,
  styles,
  anchorProps,
  anchorTarget,
  ...textProps
}: MarkdownParagraphProps) {
  const anchorStyles = styles?.anchor
  const textStyles = styles?.root
  const paras = splitParagraphs(markdown)
  return (
    <>
      {paras.map((para, idx) => {
        return (
          <Text
            {...textProps}
            className={cx(className, classes.paragraph)}
            styles={{ root: textStyles }}
            key={idx}
          >
            <MarkdownString
              styles={{ anchor: anchorStyles }}
              anchorTarget={anchorTarget}
              anchorProps={anchorProps}
              inherit
              markdown={para}
            />
          </Text>
        )
      }, markdown)}
    </>
  )
}
