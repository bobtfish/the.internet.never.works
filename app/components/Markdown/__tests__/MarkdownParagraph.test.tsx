import renderer from 'react-test-renderer';

import { MarkdownParagraph } from '../MarkdownParagraph'

test('plain text - no markdown', () => {
    const component = renderer.create(
        <MarkdownParagraph markdown="hello world" />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})

test('with markdown', () => {
    const component = renderer.create(
        <MarkdownParagraph markdown="[a link](http://www.somesite.com/) hello world" />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})

test('paragraphs - no markdown', () => {
    const input = "foo\n\nbar"
    const component = renderer.create(
        <MarkdownParagraph markdown={input} />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})

test('paragraphs - with markdown', () => {
    const input = "foo[a link](http://www.somesite.com/)\n\nbar"
    const component = renderer.create(
        <MarkdownParagraph markdown={input} />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})