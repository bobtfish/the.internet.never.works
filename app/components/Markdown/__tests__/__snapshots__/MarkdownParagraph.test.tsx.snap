// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`paragraphs - no markdown 1`] = `
[
  <p
    className="mantine-focus-auto paragraph m_b6d8b162 mantine-Text-root"
    style={{}}
  >
    <span
      className="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      data-inherit={true}
      style={{}}
    >
      foo
    </span>
  </p>,
  <p
    className="mantine-focus-auto paragraph m_b6d8b162 mantine-Text-root"
    style={{}}
  >
    <span
      className="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      data-inherit={true}
      style={{}}
    >
      bar
    </span>
  </p>,
]
`;

exports[`paragraphs - with markdown 1`] = `
[
  <p
    className="mantine-focus-auto paragraph m_b6d8b162 mantine-Text-root"
    style={{}}
  >
    <span
      className="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      data-inherit={true}
      style={{}}
    >
      foo
      <a
        className="mantine-focus-auto m_849cf0da m_b6d8b162 mantine-Text-root mantine-Anchor-root"
        data-underline="hover"
        href="http://www.somesite.com/"
        style={{}}
        target="_self"
      >
        a link
      </a>
    </span>
  </p>,
  <p
    className="mantine-focus-auto paragraph m_b6d8b162 mantine-Text-root"
    style={{}}
  >
    <span
      className="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      data-inherit={true}
      style={{}}
    >
      bar
    </span>
  </p>,
]
`;

exports[`plain text - no markdown 1`] = `
[
  <p
    className="mantine-focus-auto paragraph m_b6d8b162 mantine-Text-root"
    style={{}}
  >
    <span
      className="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      data-inherit={true}
      style={{}}
    >
      hello world
    </span>
  </p>,
]
`;

exports[`with markdown 1`] = `
[
  <p
    className="mantine-focus-auto paragraph m_b6d8b162 mantine-Text-root"
    style={{}}
  >
    <span
      className="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      data-inherit={true}
      style={{}}
    >
      <a
        className="mantine-focus-auto m_849cf0da m_b6d8b162 mantine-Text-root mantine-Anchor-root"
        data-underline="hover"
        href="http://www.somesite.com/"
        style={{}}
        target="_self"
      >
        a link
      </a>
       hello world
    </span>
  </p>,
]
`;
